corehq/apps/reports/static/reports/js/readable_form.js

```diff
@@ -0,0 +1,66 @@
+hqDefine("reports/js/readable_form", function() {
+    function showReadable() {
+        $('.form-data-raw').hide();
+        $('.form-data-readable').show();
+    }
+
+    function showRaw() {
+        $('.form-data-readable').hide();
+        $('.form-data-raw').show();
+    }
+
+    function showSkipped(show) {
+        if (show) {
+            $('.form-data-skipped').show();
+            $('.form-data-skipped-spacer').hide();
+            $('.form-data-hidden-values').each(function () {
+                $(this).show();
+            });
+        } else {
+            $('.form-data-skipped').hide();
+            $('.form-data-skipped-spacer').show();
+            $('.form-data-hidden-values').each(function () {
+                var current = $(this).next();
+                while (current.is('.form-data-question')) {
+                    if (!current.is('.form-data-skipped')) {
+                        return;
+                    }
+                    current = current.next();
+                }
+                $(this).hide();
+            });
+        }
+    }
+
+    function attachEvents() {
+        $('.showReadable').click(showReadable);
+        $('.showRaw').click(showRaw);
+        $('.formDisplayToggle a').click(function () {
+            // activate the correct 'tab' header
+            $(this).tab('show');
+            return false;
+        });
+
+        $('.showSkippedToggle').change(function () {
+            showSkipped($(this).is(':checked'));
+        }).each(function () {
+            if (!$('.form-data-skipped').length) {
+                $(this).parent('label').hide();
+            }
+        });
+    }
+
+    function init() {
+        attachEvents();
```

Comment:
Will this attach the event handlers to each matched element multiple times since `init()` is called multiple times?

Reply:
It won't because when init is called multiple times, it's always a scenario where the new content is replacing old content. But now that I say it, this code really shouldn't assume that's how it'll always be.